src/hb-aat-layout-feat-table.hh

```diff
@@ -95,15 +94,19 @@ struct FeatureName
       len = MIN (settings_count - start_offset, *records_count);
       for (unsigned int i = 0; i < len; i++)
       {
-	records_buffer[i].is_default = exclusive && not_default &&
-				       i + start_offset == (featureFlags & IndexMask);
 	records_buffer[i].name_id = settings[start_offset + i].nameIndex;
 	records_buffer[i].setting = settings[start_offset + i].setting;
       }
-      if (exclusive && !not_default && start_offset == 0 && len != 0)
-        records_buffer[0].is_default = true;
     }
-    if (is_exclusive) *is_exclusive = exclusive;
+    if (default_setting)
+    {
+      if (exclusive)
+      {
+	if (settings_count && !not_default) *default_setting = settings[0].setting;
+	else if (not_default) *default_setting = featureFlags & IndexMask;
```

Comment:
IIUC, this would be the _index of the setting in the setting name array_, not the setting itself. So I think what you want here is more like

    *default_setting = settings[featureFlags & IndexMask].setting;

(hmm, that looks like it may want a range-check).

Reply:
Oh, you are right, added with a range check. Thanks.